pkgs/by-name/bo/bombsquad/package.nix

```diff
@@ -0,0 +1,95 @@
+{
+  lib,
+  stdenv,
+  targetPlatform,
+  fetchurl,
+  python312,
+  SDL2,
+  libvorbis,
+  openal,
+  libgcc,
+  commandLineArgs ? "",
+  makeWrapper,
+  makeDesktopItem,
+  autoPatchelfHook,
+}:
+let
+  archive =
+    {
+      x86_64-linux = {
+        name = "BombSquad_Linux_x86_64";
+        hash = "sha256-VLNO0TxI/KBj8RkpGjo9Rx40f7fuV3pK2kIoKff9sRU=";
+      };
+      aarch-64-linux = {
+        name = "BombSquad_Linux_Arm64";
+        hash = "sha256-w42qhioZ9JRm004WEKzsJ3G1u09tLuPvTy8qV3DuglI=";
+      };
+    }
+    .${targetPlatform.system} or (throw "${targetPlatform.system} is unsupported.");
+in
+stdenv.mkDerivation (finalAttrs: {
+  name = "bombsquad";
+  version = "1.7.34";
+  sourceRoot = ".";
+  src = fetchurl {
+    url = "https://files.ballistica.net/bombsquad/builds/${archive.name}_${finalAttrs.version}.tar.gz";
```

Comment:
This URL goes bad after a new release of the game. See the (newly created) discussion thread here: https://github.com/efroemling/ballistica/discussions/697

Reply:
Oh Hi!! Thanks for packaging bombsquad for the AUR. I was inspired to build this nixpkg using your `PKGBUILD` as a reference only.

URL going bad is a fair issue. Lets wait for Eric to respond on the discussion. In case he doesn't we'll go ahead & just publish this as-is for now.

Btw, could you provide insights on how often you have to update the pkg version in your AUR build?

Reply:
Hi! 
> Btw, could you provide insights on how often you have to update the pkg version in your AUR build?

I've updated the AUR package a bunch, though I've also missed a lot of releases.. :)
https://aur.archlinux.org/cgit/aur.git/log/?h=bombsquad

As far as I know, Nixpkgs has cool automatic infrastructure to create update PRs.
Something like this might work:
```nix
  passthru.updateScript = genericUpdater {
    versionLister = writeShellScript "bombsquad-versionLister" ''
      curl -sL "https://files.ballistica.net/bombsquad/builds/CHANGELOG.md" \
        | grep -oP '^### \K\d+\.\d+\.\d+' \
        | head -n 1
    '';
  };
```
(^ adapted from https://github.com/NixOS/nixpkgs/blob/068c0e3c956eb0738beb212b6331a59c4fec7df5/pkgs/applications/video/filebot/default.nix#L47-L52)

Reply:
I would suggest using `writeShellApplication` instead of `writeShellScript` as the Nixpkgs manual says this about `writeShellApplication`:

> It automatically sets the PATH of the script to contain all of the listed inputs, sets some sanity shellopts (errexit, nounset, pipefail), and checks the resulting script with [shellcheck](https://github.com/koalaman/shellcheck).

https://nixos.org/manual/nixpkgs/stable/#chap-trivial-builders

```nix
passthru.updateScript = lib.getExe (writeShellApplication {
  name = "filebot-versionLister";
  runtimeInputs = [
    curl
    sed
  ];
  text = ''
    curl -s https://www.filebot.net \ 
          | sed -rne 's,^.*FileBot_([0-9]*\.[0-9]+\.[0-9]+)-portable.tar.xz.*,\1,p' 
  '';
});
```

Reply:
Added the `updateScript` :rocket: